api/v1/mongodbcommunity_types.go

```diff
@@ -497,6 +505,30 @@ func (m *MongoDBCommunity) StatefulSetReplicasThisReconciliation() int {
 	return scale.ReplicasThisReconciliation(m)
 }
 
+// GetUpdateStrategyType returns the type of RollingUpgradeStrategy that the
+// MongoDB StatefulSet should be configured with.
+func (m MongoDBCommunity) GetUpdateStrategyType() appsv1.StatefulSetUpdateStrategyType {
+	if !m.IsChangingVersion() {
+		return appsv1.RollingUpdateStatefulSetStrategyType
+	}
+	return appsv1.OnDeleteStatefulSetStrategyType
+}
+
+// IsChangingVersion returns true if an attempted version change is occurring.
+func (m MongoDBCommunity) IsChangingVersion() bool {
+	prevVersion := m.getPreviousVersion()
+	return prevVersion != "" && prevVersion != m.Spec.Version
+}
+
+// GetPreviousVersion returns the last MDB version the statefulset was configured with.
+func (m MongoDBCommunity) getPreviousVersion() string {
+	return annotations.GetLastAppliedMongoDBVersion(&m)
```

Comment:
Can we just use the `GetAnnotation` method instead of having this indirection? (Of course, after making getAnnotation public).

Reply:
yeah you are right, that's clearer! will do